### ISSUE TITLE
Batching Along New Dimensions throws RuntimeError

### ISSUE THREAD

**Issue by JeanKaddour** [rusty1s/pytorch_geometric#2108]
## 🐛 Bug

Batching along new dimensions, as explained [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/batching.html), throws the following RuntimeError:

> RuntimeError: Please look up dimensions by name, got: name = None.

## To Reproduce

```
from torch_geometric.data import Data, DataLoader
import torch

class MyData(Data):
   def __cat_dim__(self, key, item):
       if key == 'foo':
           return None
       else:
           return super().__cat_dim__(key, item)

edge_index = torch.tensor([
   [0, 1, 1, 2],
   [1, 0, 2, 1],
])
foo = torch.randn(16)

data = MyData(edge_index=edge_index, foo=foo)
data_list = [data, data]
loader = DataLoader(data_list, batch_size=2)
batch = next(iter(loader))
```

## Expected behavior

A list of attributes of shape `[num_features]` should be returned as `[num_examples, num_features]`

```
print(batch)
>>> Batch(edge_index=[2, 8], foo=[2, 16])
```

## Environment

* OS: macOS
* Python version: 3.8.5
* PyTorch version: 1.7.1
* PyTorch Geometric version: 1.6.3

**Comment by rusty1s** [comment]
This feature is not available in 1.6.3. It was integrated in one of the last commits.

**Comment by JeanKaddour** [comment]
Thanks a lot for that rapid response, Matthias! 

Will install from the source and check again.
